projects/plugins/jetpack/extensions/blocks/eventbrite/test/fixtures/jetpack__eventbrite-modal__deprecated-1.serialized.html

```diff
@@ -0,0 +1,3 @@
+<!-- wp:jetpack/eventbrite {"url":"https://www.eventbrite.com.au/e/food-hub-unsw-tickets-144144099883","eventId":144144099883,"style":"modal"} -->
+<div class="wp-block-jetpack-eventbrite"></div>
```

Comment:
It looks like the v1 deprecation is missing a migration to migrate the button to a `jetpack-button` block. This serialized output is missing the following from within the `div`:

```
<!-- wp:jetpack/button {"element":"a","uniqueId":"eventbrite-widget-id","text":"Register"} /-->
```

Reply:
Thanks @andrewserong! I'll take a look this afternoon. 

I'll grok deprecations before the end of the century, I'm sure. 😄

Reply:
So in aa060ec2bd1e5965c5532dd68c8d6636723c247f I copied the migrate function from v2 deprecation to v1. The only change was the omission of the deprecated attribute `useModal`.

It made me think: v1 and v2 deprecations are identical, save for the `useModal` attribute. 

Is it worth trying to combine them into one deprecation, and migrating the `userModal` attribute if found? 

I don't know the answer, but it seems like a huge duplication.

Reply:
Great point. I agree, I think it’s worth seeing if they can be combined to reduce the duplication, given that they’re so similar.

Reply:
Taking a look at these deprecations again, I think I don't quite understand what the difference is between the original markup that would've been generated by v1 vs v2. At the moment the source HTML for each of these deprecation fixtures (using the modal) looks to be the same.

If the markup expected by each of the deprecations is identical, then we likely don't need two separate deprecations. If it is different, then we might need to update the two source deprecation fixtures so that each of those deprecations is targeted differently? (The files: `jetpack__eventbrite-modal__deprecated-1.html` and `jetpack__eventbrite-modal__deprecated-2.html`) 🤔

Reply:
> Taking a look at these deprecations again, I think I don't quite understand what the difference is between the original markup that would've been generated by v1 vs v2.

Yeah, it's a bit confusing. I can't really see, aside from `attributes.useModal` where they differ now. 

The last deprecation was introduced in https://github.com/Automattic/jetpack/pull/15918 I think. There might be some clues there.

Reply:
After reading over the deprecations, I think I'm starting to get a clearer picture of what's needed. That said, I haven't tried tweaking the deprecations yet so take the following with a grain of salt.

My understanding is:
- V1 and v2 deprecations would match the same HTML markup whether that is for the modal or inline displays.
- The difference between v1 and v2 was the switch from `useModal` to `style`, no markup change.
- Because there was no change in markup between v1 and v2 we should be able to merge the deprecations
    - the `migrate` function will just need to handle the two cases; with `useModal` or not.
    - when v2 was added v1 should have migrated to the button block so they both produced the same result.
- Due to the fork in rendering we could still have non-modal style markup matching so need the `isEligible` check.
- It also looks to me like a deprecation may have been missed when the non-modal markup was changed in https://github.com/Automattic/jetpack/pull/14558.
    - Given the low insertion rate of the block and the small number of months between the above change and v2 I think we can ignore that.
- Even if we merge the deprecations, we'll want the fixtures to still cover the same possibilities.
    - Hopefully with some clear naming and comments to help our future selves 🙂 
- Reducing the number of deprecations does give a tiny performance bump as well.

Feel free to point out whatever I might have missed.

To move forward my suggestion would be to:
1. Merge v1 & v2
2. Update the fixtures as needed to cover each use-case the deprecations handle
3. Document what's going on inline as much as possible

Reply:
Thanks for taking the time to review and confirm all this @aaronrobertshaw. 🙇 

I've made an attempt at this in 7b9d77a3649df8acf0721ddd7ad804de4ecae790

If folks think it's all too much noise for this PR I can extract it and do a follow up.